src/main/kotlin/org/radarbase/push/integration/common/auth/Oauth1Signing.kt

```diff
@@ -86,16 +81,12 @@ class Oauth1Signing(
     private fun String.toBytesUtf8() = this.toByteArray()
 
     private fun HashMap<String, String>.toHeaderFormat() =
-        filterKeys { it in baseKeys }
-            .toMutableList()
-            .sortWith { (key, _) -> key }  // sort in place
-            .joinToString(", ") { (key, value) -> "$key=\"$value\"" }
-
-    private fun HashMap<String, String>.encodeForSignature() =
-        toMutableList()
-            .sortWith { (key, _) -> key }  // sort in place
-            .joinToString("&") { (key, value) -> "$key=$value" }
-            .encodeUtf8()
+        filter { it.key in baseKeys }
+            .toList()
+            .sortedBy { (key, _) -> key }
+            .toMap()
+            .map { "${it.key}=\"${it.value}\"" }
+            .joinToString(", ")
```

Comment:
Why use the old implementation instead of 
```
        filterKeys { it in baseKeys }
             .toMutableList()
             .sortWith { (key, _) -> key }  // sort in place
             .joinToString(", ") { (key, value) -> "$key=\"$value\"" }
```

Reply:
In `toMutableList()` I was getting `Unresolved reference. None of the following candidates is applicable because of receiver type mismatch: ` ? I switched it back in the meantime

Reply:
Ah you're right, in this case it should be

```
filterKeys { it in baseKeys }
             .entries
             .toMutableList()
             .sortWith { (key, _) -> key }  // sort in place
             .joinToString(", ") { (key, value) -> "$key=\"$value\"" }
```

Reply:
okay thanks